### PR TITLE
CI/STYLE: Add auto-fix to ruff

### LINKED ISSUE #51814: STYLE: Add auto fix to ruff
How about add fix/autofix ruff?

I recently worked on [STYLE enable ruff TCH](https://github.com/pandas-dev/pandas/issues/51740) and found that ruff reports some error could (eg: Trailing whitespace, ) been easily autofixed, by adding `--fix` (or more specific flags) as argument to config. 

I must admit that these errors can also be fixed manually. However, if the automation does not significantly slow down the process, why not let them be fixed automatically? It is worth noting that automatic fixes are still being developed and modified, and whether they should be used or not may require further discussion.

Some maybe related:
[ruff-tried-to-fix-something-but-it-broke-my-code?](https://beta.ruff.rs/docs/faq/#ruff-tried-to-fix-something-but-it-broke-my-code)
[Usage - Ruff](https://beta.ruff.rs/docs/usage/)
[Configuration - Ruff](https://beta.ruff.rs/docs/configuration/#command-line-interface)

### PR DESCRIPTION
**Pull request by luke396**
- [x] closes #51814 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by luke396** [comment]
It seems that `ruff`'s `fix` are duplicates of some of our existing autofix. Maybe we need to make some choices between existing methods and `ruff`?

**Comment by MarcoGorelli** [comment]
> It seems that ruff's fix are duplicates of some of our existing autofix. Maybe we need to make some choices between existing methods and ruff?

I don't undertstand what you mean, sorry, could you clarify please?

You'll also need to add `exit-non-zero-on-fix` (check the ruff docs for the exact name, it's something like that)

**Comment by luke396** [comment]
> I don't undertstand what you mean, sorry, could you clarify please?
> 
> You'll also need to add `exit-non-zero-on-fix` (check the ruff docs for the exact name, it's something like that)

I have noticed that some of the rules that `ruff` automatically fixes may be duplicates of the rules we have already added to the main branch. However, this is just an intuitive observation, and I have not confirmed it further.

And `--exit-non-zero-on-fix` has added.

File: pyproject.toml [hunk 1]
```diff
@@ -197,6 +197,8 @@ exclude = '''
 line-length = 88
 update-check = false
 target-version = "py38"
+fix-only = true
```
Comment (MarcoGorelli) [inline pyproject.toml:200]:
```suggestion
fix = true
```

**Review by MarcoGorelli — APPROVED** [review]
thanks @luke396

**Pull request merged**